### PR TITLE
move from bash to snapshot tests in rspec

### PR DESCRIPTION
**Pull request by michaelglass**
tests a successful abq run and moves abq output testing to rspec

addresses #44

### CODE REVIEW

**Review by pierrebeaucamp — APPROVED** [review]

**Review by pierrebeaucamp — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -216,11 +220,10 @@ jobs:
       - name: start abq work
         # abq work _should_ fail because we have failing tests, but we want to ensure it only has a single line of output
         run: |
-          set +e
-          echo "run worker"
-          abq work -n cpu-cores --run-id "$GITHUB_RUN_ID-$GITHUB_RUN_ATTEMPT-${{matrix.ruby}}-${{matrix.gemfile}}" 2>&1 |
-            bin/test-worker-output.rb
+          bin/test-worker-output.rb success
```
Comment (michaelglass) [inline .github/workflows/ci.yml:223]: could run this as a matrix but it's such a pain to update the matrix that I'll punt for now.

If / when we set required status via a script or terraform, we'll expand this to a matrix.

**Comment by dan-manges** [comment]
These snapshot test are nice :clap:

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -216,11 +220,10 @@ jobs:
...
+          bin/test-worker-output.rb failure
```
Comment (dan-manges) [inline .github/workflows/ci.yml:224]: I'm confused about what's going on here. Why are we calling both

```
bin/test-worker-output.rb success
bin/test-worker-output.rb failure
```

Is there a test that's dynamic that is passing or failing based on the same input?

**Comment by dan-manges** [comment]
I'd like to see us make the underlying tests more intentional. I don't think they're explicit enough to indicate which test is in place to test which behavior. For example, I'm trying to figure out where we're testing failures, and I see this test

```
  it 'example 1 with tags', :foo, :bar do
    expect(false).to eq(true)
  end
```

and I think this is there to test failing tests, but according to the description it looks like it's there to test tags.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (michaelglass) [reply .github/workflows/ci.yml:224]: should be cleared up in #48

**Comment by michaelglass** [comment]
@dan-manges yah agreed. First I want to make it easy to write integration tests, then I want to write better integration tests

**Pull request merged**